Bonsai.Tinkerforge/BrickletGPSV2.cs

```diff
@@ -0,0 +1,211 @@
+                        device.AltitudeCallback -= handler;
+                    });
+                });
+            });
+
+            // DateTime stream
+            IObservable<DateTimeData> dateTimeStream = source.SelectStream(connection =>
+            {
+                var device = new global::Tinkerforge.BrickletGPSV2(Uid, connection);
+                connection.Connected += (sender, e) =>
+                {
+                    device.SetDateTimeCallbackPeriod(Period);
+                };
+
+                return Observable.Create<DateTimeData>(observer =>
+                {
+                    observer.OnNext(new DateTimeData()); // Initialize an empty data struct in case one of the data providers doesn't return anything to the stream combination
+
+                    global::Tinkerforge.BrickletGPSV2.DateTimeEventHandler handler = (sender, date, time) =>
+                    {
+                        observer.OnNext(new DateTimeData(date, time));
+                    };
+
+                    device.DateTimeCallback += handler;
+                    return Disposable.Create(() =>
+                    {
+                        try { device.SetDateTimeCallbackPeriod(0); }
+                        catch (NotConnectedException) { }
+                        device.DateTimeCallback -= handler;
+                    });
+                });
+            });
+
+            // Coordinate stream
+            IObservable<CoordinateData> coordinateStream = source.SelectStream(connection =>
+            {
+                var device = new global::Tinkerforge.BrickletGPSV2(Uid, connection);
+                connection.Connected += (sender, e) =>
+                {
+                    device.SetStatusLEDConfig((byte)StatusLed);
+                    device.SetSBASConfig((byte)SBAS);
+                    device.SetCoordinatesCallbackPeriod(Period);
+                };
+
+                return Observable.Create<CoordinateData>(observer =>
+                {
+                    observer.OnNext(new CoordinateData()); // Initialize an empty data struct in case one of the data providers doesn't return anything to the stream combination
+
+                    global::Tinkerforge.BrickletGPSV2.CoordinatesEventHandler handler = (sender, latitude, ns, longitude, ew) =>
+                    {
+                        observer.OnNext(new CoordinateData(latitude, longitude, ns, ew));
+                    };
+
+                    device.CoordinatesCallback += handler;
+                    return Disposable.Create(() =>
+                    {
+                        try { device.SetCoordinatesCallbackPeriod(0); }
+                        catch (NotConnectedException) { }
+                        device.CoordinatesCallback -= handler;
+                    });
+                });
+            });
+
+            return statusStream.CombineLatest(coordinateStream, altitudeStream, dateTimeStream, (s1, s2, s3, s4) => Tuple.Create(s1, s2, s3, s4));
+        }
```

Comment:
Assuming that reusing the device UID works for the same IPConnection, a totally different, but maybe more flexible approach could be to simply have one operator for each GPSV2 data stream, i.e. literally one node for Altitude, one node for Coordinates, one node for Status, etc.

That would give us maximum flexibility to downstream combine everything as we want, or to just use each stream independently, etc.

Reply:
If we go this route, it might be useful to create a helper abstract class to contain common functionality such as status code, etc and just override to pick which stream you want to return.

Reply:
So far it doesn't appear that the same UID can be reused easily, but I will investigate further